src/NLog/LogFactory.cs

```diff
@@ -1386,6 +1408,26 @@ private IEnumerable<Logger> GetLoggers()
 
                 return values;
             }
+
+            public void PruneLoggerCache()
```

Comment:
We're removing the "guid" like loggers here, isn;t?

Reply:
Yes when chaining multiple WithProperty-calls, then it will leave several loggers entries. If loggers has no static references, then it will slowly fill up the logger-cache. Only doing the prune, when activating the context-feature.